### PR TITLE
FEATURE - Added commit function to sparkswap exchange

### PR DESCRIPTION
**Pull request by dannypaz**
This PR adds functionality for `commit` and adds an example showing users how to commit funds to the sparkswap exchange

### CODE REVIEW

**Comment by dannypaz** [comment]
[Create a `commit-funds` command for CCXT (custom command)](https://trello.com/c/DwqtyACV/687-create-a-commit-funds-command-for-ccxt-custom-command)

File: js/sparkswap.js [hunk 1]
```diff
@@ -200,8 +210,22 @@ module.exports = class sparkswap extends Exchange {
         throw new ExchangeError ('Not Implemented');
     }
 
-    async commit () {
-        throw new ExchangeError ('Not Implemented');
+    async commit (symbol = '', balance = '', market = '', params = {}) {
```
Comment (martineehrlich) [inline js/sparkswap.js]: I think we need another name for symbol because ccxt symbol === marketName. Maybe currency?

File: js/sparkswap.js [hunk 2]
```diff
@@ -200,8 +210,22 @@ module.exports = class sparkswap extends Exchange {
...
+        let limit = this.safeFloat (this.channelLimits, symbol);
+        if (limit < parseFloat (balance)) {
```
Comment (martineehrlich) [inline js/sparkswap.js:215]: We don't want to give them an error that says the limit is too big? I guess I have always felt weird about changing the amount they want to commit without confirmation that this is what they want to do?

**Review by martineehrlich — CHANGES_REQUESTED** [review]

File: js/sparkswap.js (hunk 1 above)
Comment (dannypaz) [reply js/sparkswap.js]: @martineehrlich good catch. In their examples they use `code`.

File: js/sparkswap.js (hunk 2 above)
Comment (dannypaz) [reply js/sparkswap.js:215]: This was updated. we now throw an error if the balance is above the limit

File: js/sparkswap.js [hunk 3]
```diff
@@ -210,19 +210,13 @@ module.exports = class sparkswap extends Exchange {
...
+        let limit = this.safeFloat (this.channelLimits, code);
         if (limit < parseFloat (balance)) {
-            // We automatically set the limit as the balance, if the balance is greater
-            // than the allowed channel limit specified by currency. This is so that
-            // the user will not have to implement logic themselves to handle if the commit
-            // has failed due to limits.
-            //
-            // This functionality will be removed once sparkswap is on MainNet
-            balance = limit;
+            throw new BadRequest ('Balance does not meet limit for currency code: ' + code + ' balance: ' + balance + ' limit: ' + limit);
```
Comment (martineehrlich) [inline js/sparkswap.js]: Balance exceeds channel limit for currency, the maximum balance you can commit to a code channel is _______.

Comment (dannypaz) [reply js/sparkswap.js]: +1. thanks for the suggestion.

Example will be: ```Failed to commit balance: Balance exceeds channel limit for currency, the maximum balance you can commit for BTC is: 0.16777215```

**Review by martineehrlich — APPROVED** [review]
LGTM

**Pull request merged**